### PR TITLE
Add rule to convert short template notation to bundle notation

### PR DESCRIPTION
**Pull request by arnaubuch**
Converts short Twig template notation (deprecated in 5.0) for bundle notation with `@` prefix.

Needed this rule for my own purposes but tought I'd share for anyone that might need it.

### CODE REVIEW

File: src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bundle/FrameworkBundle/Templating/TemplateNameParser.php
+ * @see https://symfony.com/doc/4.4/templates.html#bundle-templates
+ *
+ * @see \Rector\Symfony\Tests\Rector\ClassMethod\TemplateAnnotationToThisRenderRector\TemplateAnnotationToThisRenderRectorTest
+ */
+
```
Comment (johanadivare) [inline src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php:24]:
```suggestion
```

File: src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\ThisType;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bundle/FrameworkBundle/Templating/TemplateNameParser.php
+ * @see https://symfony.com/doc/4.4/templates.html#bundle-templates
```
Comment (johanadivare) [inline src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php:20]: Maybe you can also add a `@see https://github.com/symfony/symfony/pull/21035`

File: src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            ! $this->nodeNameResolver->isName($node->name, 'renderView')) {
+            return null;
+        }
+
+        $objectType = $this->nodeTypeResolver->getType($node->var);
+        if (! $objectType instanceof ThisType) {
+            return null;
+        }
+
+        $controllerType = new ObjectType('Symfony\Bundle\FrameworkBundle\Controller\Controller');
```
Comment (johanadivare) [inline src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php:78]: I would check this before the `ThisType` check above

File: src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bundle/FrameworkBundle/Templating/TemplateNameParser.php
+ * @see https://symfony.com/doc/4.4/templates.html#bundle-templates
+ *
+ * @see \Rector\Symfony\Tests\Rector\ClassMethod\TemplateAnnotationToThisRenderRector\TemplateAnnotationToThisRenderRectorTest
+ */
+
+final class ConvertRenderTemplateShortNotationToBundleSyntaxRector extends AbstractRector
```
Comment (johanadivare) [inline src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php:25]: Might be nice to add this rule in symfony43.php as its fixing a deprecation there or should it be in symfony50.php as it is removed in there?

File: tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_skip_fixture.php.inc [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+class BaseController extends Controller
+{
+    function indexAction()
+    {
+        $this->render('@app/Landing/Main/index.html.twig');
+    }
+}
+
+?>
+-----
```
Comment (johanadivare) [inline tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_skip_fixture.php.inc:16]: With a skip test you can remove the bottom part

**Comment by arnaubuch** [comment]
@johanadivare Applied suggestions, waiting for another review!

File: src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        $objectType = $this->nodeTypeResolver->getType($node->var);
+        $controllerType = new ObjectType('Symfony\Bundle\FrameworkBundle\Controller\Controller');
+        if (! $controllerType->isSuperTypeOf($objectType)->yes()) {
+            return null;
+        }
+
+        if (! $objectType instanceof ThisType) {
+            return null;
+        }
+
```
Comment (samsonasik) [inline src/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector.php:82]: First class callable `$this->call(...)` need to be skipped as well.

```suggestion
        if ($node->isFirstClassCallable()) {
               return null;
        }
```

**Comment by samsonasik** [comment]
Please run:

```
vendor/bin/rector && composer fix-cs 
```

to make CI green.

**Review by samsonasik — APPROVED** [review]

File: tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_skip_fixture.php.inc [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Rector\Symfony\Tests\Rector\MethodCall\ConvertRenderTemplateShortNotationToBundleSyntaxRector\Fixture;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+class BaseController extends Controller
```
Comment (JohJohan) [inline tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_skip_fixture.php.inc:7]: Name `skip_already_bundle_name.php.inc` and the controller `SkipAlreadyBundleName`

File: tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_fixture.php.inc [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rector\Symfony\Tests\Rector\MethodCall\ConvertRenderTemplateShortNotationToBundleSyntaxRector\Fixture;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+class BaseController extends Controller
```
Comment (JohJohan) [inline tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_fixture.php.inc:7]: You could rename the fixture file `shortToBundleName.php.inc` and controller to `ShortToBundleName` to make clear what this fixture is changing, maybe you can find better naming

File: tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_fixture_renderView.php.inc [hunk 9]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rector\Symfony\Tests\Rector\MethodCall\ConvertRenderTemplateShortNotationToBundleSyntaxRector\Fixture;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+class BaseController extends Controller
```
Comment (JohJohan) [inline tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_fixture_renderView.php.inc:7]: You could rename the fixture file `shortToBundleNameRenderView.php.inc` and controller to `ShortToBundleNameRenderView` to make clear what this fixture is changing, maybe you can find better naming

File: tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_skip_fixture.php.inc (hunk 7 above)
Comment (arnaubuch) [reply tests/Rector/MethodCall/ConvertRenderTemplateShortNotationToBundleSyntaxRector/Fixture/test_skip_fixture.php.inc:7]: Renamed the three tests, hopefully it's clear enough.

**Comment by TomasVotruba** [comment]
Looks good, thank you :+1:

**Pull request merged**